sdk/hash/src/lib.rs

```diff
@@ -64,15 +68,25 @@ impl AsRef<[u8]> for Hash {
     }
 }
 
+fn to_base58(h: &Hash, f: &mut fmt::Formatter) -> fmt::Result {
+    let mut out = [0u8; MAX_BASE58_LEN];
+    let out_slice: &mut [u8] = &mut out;
+    // This will never fail because the only possible error is BufferTooSmall,
+    // and we will never call it with too small a buffer.
+    let len = bs58::encode(h.0).onto(out_slice).unwrap();
```

Comment:
it's so strange that bs58 doesn't implement `onto` for a byte array, but oh well!